frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/ClusterActionDropdown.tsx

```diff
@@ -114,16 +114,20 @@ export function ClusterActionDropdown(props: { cluster: Cluster; isKebab: boolea
             rbacCreate(SecretDefinition, cluster.namespace),
           ],
         },
-        {
-          id: ClusterAction.RemoveAutomationTemplate,
-          text: t('Remove automation template'),
-          click: () => setShowRemoveAutomationModal(true),
-          isAriaDisabled: true,
-          rbac: [
-            rbacDelete(ClusterCuratorDefinition, cluster.namespace),
-            rbacDelete(SecretDefinition, cluster.namespace),
-          ],
-        },
+        ...(cluster.hasAutomationTemplates
+          ? [
+              {
+                id: ClusterAction.RemoveAutomationTemplate,
+                text: t('Remove automation template'),
+                click: () => setShowRemoveAutomationModal(true),
+                isAriaDisabled: true,
+                rbac: [
+                  rbacDelete(ClusterCuratorDefinition, cluster.namespace),
+                  rbacDelete(SecretDefinition, cluster.namespace),
+                ],
+              },
+            ]
+          : []),
```

Comment:
This array is already filtered by the `clusterSupportsAction` function. If you use it, the logic can be centralized and reused where needed.